lightning/src/blinded_path/mod.rs

```diff
@@ -75,25 +76,32 @@ impl BlindedPath {
 		})
 	}
 
-	/// Create a blinded path for a payment, to be forwarded along `path`. The last node
-	/// in `path` will be the destination node.
+	/// Create a blinded path for a payment, to be forwarded along `intermediate_nodes`, where each
+	/// node is composed of `(node_id, tlvs, htlc_maximum_msat)`.
+	///
+	/// Errors if:
+	/// * a provided node id is invalid
+	/// * [`BlindedPayInfo`] calculation results in an integer overflow
+	/// * any unknown features are required in the provided [`ForwardTlvs`]
 	///
-	/// Errors if `path` is empty or a node id in `path` is invalid.
+	/// [`ForwardTlvs`]: crate::blinded_path::payment::ForwardTlvs
 	//  TODO: make all payloads the same size with padding + add dummy hops
 	pub fn new_for_payment<ES: EntropySource, T: secp256k1::Signing + secp256k1::Verification>(
-		intermediate_nodes: &[(PublicKey, payment::ForwardTlvs)], payee_node_id: PublicKey,
-		payee_tlvs: payment::ReceiveTlvs, entropy_source: &ES, secp_ctx: &Secp256k1<T>
-	) -> Result<Self, ()> {
+		intermediate_nodes: &[(PublicKey, payment::ForwardTlvs, u64)], payee_node_id: PublicKey,
```

Comment:
For reviewers: before I rebase #2540 on this, does this interface for `intermediate_nodes` seem acceptable? We could make a struct to contain the 3 tuple values.

Reply:
Sure, I'll put it in a struct. I suspect @jkczyz would prefer that too.

Reply:
Yeah, a struct SGTM.